### PR TITLE
feat: Run jobs using system executor

### PR DESCRIPTION
**Pull request by Ashish-devtron**
# Description

Fixes https://github.com/devtron-labs/devtron/issues/4205

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/pipeline/CiHandler.go [hunk 1]
```diff
@@ -645,14 +647,32 @@ func (impl *CiHandlerImpl) CancelBuild(workflowId int) (int, error) {
...
+			impl.Logger.Errorw("error in getting rest config by cluster id", "err", err)
+			return 0, err
+		}
+	}
+
+	//runningWf, err := impl.workflowService.GetWorkflow(workflow.Name, workflow.Namespace, isExt, env)
+	//if err != nil {
+	//	impl.Logger.Errorw("cannot find workflow ", "err", err)
+	//	return 0, errors.New("cannot find workflow " + workflow.Name)
+	//}
```
Comment (subhashish-devtron) [inline pkg/pipeline/CiHandler.go:673]: clean

File: pkg/pipeline/ArgoWorkflowExecutor.go [hunk 2]
```diff
@@ -135,6 +136,21 @@ func (impl *ArgoWorkflowExecutorImpl) ExecuteWorkflow(workflowTemplate bean.Work
...
+
+	wfClient, err := impl.getClientInstance(namespace, clusterConfig)
+	if err != nil {
+		impl.logger.Errorw("cannot build wf client", "wfName", workflowName, "err", err)
+		return nil, err
+	}
+	wf, err := wfClient.Get(context.Background(), workflowName, v1.GetOptions{})
+	if err != nil {
+		impl.logger.Errorw("cannot find workflow", "name", workflowName, "err", err)
+		return nil, errors.New("cannot find workflow " + workflowName)
```
Comment (subhashish-devtron) [inline pkg/pipeline/ArgoWorkflowExecutor.go:149]: fmt.errorf

File: pkg/pipeline/CiHandler.go (hunk 1 above)
Comment (subhashish-devtron) [inline pkg/pipeline/CiHandler.go:673]: clean

File: pkg/pipeline/CiHandler.go [hunk 3]
```diff
@@ -645,14 +647,32 @@ func (impl *CiHandlerImpl) CancelBuild(workflowId int) (int, error) {
...
+		return 0, err
 	}
+	var restConfig *rest.Config
+	if isExt {
+		restConfig, err = impl.K8sUtil.GetRestConfigByCluster(clusterConfig)
+		if err != nil {
+			impl.Logger.Errorw("error in getting rest config by cluster id", "err", err)
+			return 0, err
+		}
+	}
```
Comment (subhashish-devtron) [inline pkg/pipeline/CiHandler.go:666]: correct this as discussed offlin

File: pkg/pipeline/CiHandler.go [hunk 4]
```diff
@@ -1606,8 +1626,26 @@ func (impl *CiHandlerImpl) UpdateCiWorkflowStatusFailure(timeoutForFailureCiBuil
...
+				return err
+			}
+			var restConfig *rest.Config
+			if isExt {
+				restConfig, err = impl.K8sUtil.GetRestConfigByCluster(clusterConfig)
+				if err != nil {
+					impl.Logger.Errorw("error in getting rest config by cluster id", "err", err)
+					return err
+				}
+			}
```
Comment (subhashish-devtron) [inline pkg/pipeline/CiHandler.go:1646]: this logic is duplicated and does not belong here

File: pkg/pipeline/CiHandler.go [hunk 5]
```diff
@@ -1643,10 +1681,15 @@ func (impl *CiHandlerImpl) UpdateCiWorkflowStatusFailure(timeoutForFailureCiBuil
 						// skip this and process for next ci workflow
 					}
 				}
-
-				//check workflow status,get the status
-				if wf.Status.Phase == v1alpha1.WorkflowFailed && wf.Status.Message == POD_DELETED_MESSAGE {
-					isPodDeleted = true
+				if ciWorkflow.ExecutorType == pipelineConfig.WORKFLOW_EXECUTOR_TYPE_AWF {
+					createdWf := &v1alpha1.Workflow{}
+					obj := wf.Items[0].Object
```
Comment (subhashish-devtron) [inline pkg/pipeline/CiHandler.go:1686]: will this not panic?

File: pkg/pipeline/ArgoWorkflowExecutor.go [hunk 6]
```diff
@@ -135,6 +137,40 @@ func (impl *ArgoWorkflowExecutorImpl) ExecuteWorkflow(workflowTemplate bean.Work
...
+	wfClient, err := impl.getClientInstance(namespace, clusterConfig)
+	if err != nil {
+		impl.logger.Errorw("cannot build wf client", "wfName", workflowName, "err", err)
+		return nil, err
+	}
+	wf, err := wfClient.Get(context.Background(), workflowName, v1.GetOptions{})
+	if err != nil {
+		impl.logger.Errorw("cannot find workflow", "name", workflowName, "err", err)
+		return nil, fmt.Errorf("cannot find workflow %s", workflowName)
+	}
```
Comment (kripanshdevtron) [inline pkg/pipeline/ArgoWorkflowExecutor.go:166]: duplicate code

**Review by subhashish-devtron — DISMISSED** [review]

**Review by kripanshdevtron — APPROVED** [review]

**Pull request merged**